torchgeo/datasets/biomassters.py

```diff
@@ -0,0 +1,344 @@
+
+
+class BioMassters(NonGeoDataset):
+    """BioMassters Dataset for Aboveground Biomass prediction.
+
+    Dataset intented for Aboveground Biomass (AGB) prediction
+    over Finnish forests based on Sentinel 1 and 2 data with
+    corresponding target AGB mask values generated by Light Detection
+    and Ranging (LiDAR).
+
+    Dataset Format:
+
+    * .tif files for Sentinel 1 and 2 data
+    * .tif file for pixel wise AGB target mask
+    * .csv files for metadata regarding features and targets
+
+    Dataset Features:
+
+    * 13,000 target AGB masks of size (256x256px)
+    * 12 months of data per target mask
+    * Sentinel 1 and Sentinel 2 data for each location
+    * Sentinel 1 (specify channels here) available for each month
+    * Sentinel 2 (B02, B03, B04, B05, B06, B07, B08, B8A, B11, B12)
+      not available for each month due to ESA aquisition halt over the region
+      during particular periods
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * TBD
+
+    .. versionadded:: 0.5
+    """
+
+    valid_splits = ["train", "test"]
+    valid_sensors = ["S1", "S2"]
+
+    url = "https://huggingface.co/datasets/nascetti-a/BioMassters/resolve/main/{}"
+
+    md5s = {"train_features": "123", "test_features": "456", "train_agbm": "789"}
+
+    metadata_filename = "The_BioMassters_-_features_metadata.csv.csv"
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "train",
+        sensors: list[str] = ["S1", "S2"],
+        as_time_series: bool = False,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new instance of BioMassters dataset.
+
+        Args:
+            root: root directory where dataset can be found
+            split: train or test split
+            sensors: which sensors to consider for the sample
+            as_time_series: whether or not to return all available
+                time-steps or just a single one for a given target location
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        RuntimeError:
+            AssertionError: if ``split`` or ``sensors`` is invalid
+            ImportError: if pandas is not installed
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        self.root = root
+
+        assert (
+            split in self.valid_splits
+        ), f"Please choose one of the valid splits {self.valid_splits}."
+        self.split = split
+
+        assert set(sensors).issubset(
+            set(self.valid_sensors)
+        ), f"Please choose a subset of valid sensors {self.valid_sensors}."
+        self.sensors = sensors
+        self.as_time_series = as_time_series
+        self.download = download
+        self.checksum = checksum
+
+        try:
```

Comment:
See #1586, we can merge these in either order

Reply:
#1586 has been merged, pandas is now a required dep so you can simplify a lot of this.